core/trino-main/src/test/java/io/trino/operator/TestHashAggregationOperator.java

```diff
@@ -698,6 +713,77 @@ public void testMemoryTracking()
+                Optional.empty(),
+                Optional.empty(),
+                100,
+                Optional.of(DataSize.ofBytes(1)),
+                joinCompiler,
+                blockTypeOperators,
+                new PartialAggregationControlFactory(true, 5, 0.8));
+
+        DriverContext driverContext = createDriverContext(1024);
+
+        try (Operator operator = operatorFactory.createOperator(driverContext)) {
+            List<Page> input = rowPagesBuilder(false, hashChannels, BIGINT)
+                    .addSequencePage(10, 0)
+                    .addBlocksPage(createRLEBlock(1, 10))
+                    .build();
+            List<Page> expectedPages = rowPagesBuilder(BIGINT, BIGINT)
+                    .addSequencePage(10, 0, 0)
+                    .addBlocksPage(createRLEBlock(1, 10), createRLEBlock(1, 10)) // we are expecting second page with raw values
+                    .build();
+            MaterializedResult expected = resultBuilder(driverContext.getSession(), BIGINT, BIGINT)
+                    .pages(expectedPages)
+                    .build();
+
+            // Drive the operator to completion
+            List<Page> outputPages = toPages(operator, input.iterator());
+
+            MaterializedResult actual = toMaterializedResult(operator.getOperatorContext().getSession(), expected.getTypes(), outputPages);
+
+            assertEquals(actual.getTypes(), expected.getTypes());
+            assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+        }
+
+        // next operator using the same factory, reuses PartialAggregationControl, so it should only produce raw pages
+        DriverContext secondDriverContext = createDriverContext(1024);
+        try (Operator operator = operatorFactory.createOperator(secondDriverContext)) {
+            List<Page> input = rowPagesBuilder(false, hashChannels, BIGINT)
+                    .addBlocksPage(createRLEBlock(1, 10))
+                    .addBlocksPage(createRLEBlock(2, 10))
+                    .build();
+            List<Page> expectedPages = rowPagesBuilder(BIGINT, BIGINT)
+                    .addBlocksPage(createRLEBlock(1, 10), createRLEBlock(1, 10))
+                    .addBlocksPage(createRLEBlock(2, 10), createRLEBlock(2, 10))
+                    .build();
+            MaterializedResult expected = resultBuilder(secondDriverContext.getSession(), BIGINT, BIGINT)
+                    .pages(expectedPages)
+                    .build();
+
+            // Drive the operator to completion
+            List<Page> outputPages = toPages(operator, input.iterator());
+
+            MaterializedResult actual = toMaterializedResult(operator.getOperatorContext().getSession(), expected.getTypes(), outputPages);
+
+            assertEquals(actual.getTypes(), expected.getTypes());
+            assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+        }
```

Comment:
Add a test when you actually assert that the partial aggregation is being disabled after 5 rows

Reply:
do you mean `PartialAggregationControl.partialAggregationDisabled`? because other than asserting this, the above test checks that PA is being disabled (RLE blocks are not being grouped) and I don't think asserting `PartialAggregationControl.partialAggregationDisabled` is a good idea (it's a implementation detail)

Reply:
I refactored this tests + added comments as requested